examples/pytorch/graphsaint/utils.py

```diff
@@ -0,0 +1,100 @@
+import json
+import os
+from functools import namedtuple
+import scipy.sparse
+from sklearn.preprocessing import StandardScaler
+import dgl
+import numpy as np
+import torch
+from sklearn.metrics import f1_score
+
+class Logger(object):
+    '''A custom logger to log stdout to a logging file.'''
+    def __init__(self, path):
+        """Initialize the logger.
+
+        Paramters
+        ---------
+        path : str
+            The file path to be stored in.
+        """
+        self.path = path
+
+    def write(self, s):
+        with open(self.path, 'a') as f:
+            f.write(str(s))
+        print(s)
+        return
+
+
+def save_log_dir(args):
+    log_dir = './log/{}/{}'.format(args.dataset, args.note)
+    os.makedirs(log_dir, exist_ok=True)
+    return log_dir
+
+
+def calc_f1(y_true, y_pred, multitask):
+    if multitask:
+        y_pred[y_pred > 0] = 1
+        y_pred[y_pred <= 0] = 0
+    else:
+        y_pred = np.argmax(y_pred, axis=1)
+    return f1_score(y_true, y_pred, average="micro"), \
+        f1_score(y_true, y_pred, average="macro")
+
+
+def evaluate(model, g, labels, mask, multitask=False):
+    model.eval()
+    with torch.no_grad():
+        logits = model(g)
+        logits = logits[mask]
+        labels = labels[mask]
+        f1_mic, f1_mac = calc_f1(labels.cpu().numpy(),
+                                 logits.cpu().numpy(), multitask)
+        return f1_mic, f1_mac
+
+
+# load data of GraphSaint and translate them to the format of dgl
+def load_data(args, multitask):
+    prefix = "data/{}".format(args.dataset)
+    DataType = namedtuple('Dataset', ['num_classes', 'g'])
+
+    adj_full = scipy.sparse.load_npz('./{}/adj_full.npz'.format(prefix)).astype(np.bool)
+    g = dgl.from_scipy(adj_full)
```

Comment:
It seems that you are not using `'./{}/adj_train.npz'.format(prefix)`. According to the doc [here](https://github.com/GraphSAINT/GraphSAINT/blob/master/graphsaint/utils.py#L49), using this will remove the edges between the training nodes and the validation/test nodes.

Reply:
I use "self.train_g = g.subgraph(train_nid)" to generate training graph during sampling.